Consider including "-std=c99" CFLAG for Linux as well

I see it's [already there for OpenBSD](https://github.com/crawshaw/sqlite/commit/ad1523d5270088cf420807c8ba9c25c0723c76f3).

On a very old Debian box (which I use to make sure our binaries are happy with very old glibcs), `blocking_step.c` fails to build:

```
--> linux/amd64 error: exit status 2
Stderr: # github.com/itchio/butler/vendor/crawshaw.io/sqlite
blocking_step.c: In function ‘unlock_notify_cb’:
blocking_step.c:42:2: error: ‘for’ loop initial declarations are only allowed in C99 mode
blocking_step.c:42:2: note: use option -std=c99 or -std=gnu99 to compile your code
```

I'm not sure if recent GCCs default to `gnu99` or whatever, I'm happy with either:

  1. Adding the `-std=c99` cflag
  2. Making sure `.c` files in this repo are C89

--- a/sqlite.go
+++ b/sqlite.go
@@ -27,6 +27,7 @@ package sqlite
 // #cgo CFLAGS: -DSQLITE_USE_ALLOCA
 // #cgo CFLAGS: -DSQLITE_ENABLE_COLUMN_METADATA
 // #cgo linux LDFLAGS: -ldl -lm
+// #cgo linux CFLAGS: -std=c99
 // #cgo openbsd LDFLAGS: -lm
 // #cgo openbsd CFLAGS: -std=c99
 //

